### PR TITLE
SAK-23630 new unboundid ldap provider

### PR DESCRIPTION
**Pull request by ottenhoff**
I haven't touched the tests yet .... but this appears to be working for me on my DEV.... 

@buckett any interest here?

### CODE REVIEW

**Comment by ern** [comment]
Really nice to see this! The Novell library is pretty old and hasn't had any updates in a long time.
I would say lets get this in as experimental, so that it can be kicked around ;)

File: providers/unboundid/src/test/org/sakaiproject/unboundid/UnboundidProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPEntry;
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPException;
+import com.unboundid.ldap.sdk.migrate.ldapjdk.LDAPSearchResults;
+
+/**
+ * 
+ * @author dmccallum@unicon.net
+ *
+ */
+public class UnboundidDirectoryProviderTest extends MockObjectTestCase {
```
Comment (jonespm) [inline providers/unboundid/src/test/org/sakaiproject/unboundid/UnboundidProviderTest.java]: This test is failing as it doesn't match the filename
The test is failing here with an incorrectly named file 
[ERROR] COMPILATION ERROR : 
[ERROR] /home/travis/build/sakaiproject/sakai/providers/unboundid/src/test/org/sakaiproject/unboundid/UnboundidProviderTest.java:[51,8] class UnboundidDirectoryProviderTest is public, should be declared in a file named UnboundidDirectoryProviderTest.java

**Comment by buckett** [comment]
Yeah, I'd be really interested. Just haven't had time to look yet.

**Comment by marktriggs** [comment]
I've pulled these patches into NYU's Sakai 11 branch to give them a spin.  I added SSL Socket support back in for NYU here:

https://github.com/NYUeServ/sakai11/commit/41a6a8a2ef47ee5fc2999b3b85e8a501e01d8d63

I can send a proper PR if/when this one is merged.

**Comment by jonespm** [comment]
Was this still in progress? Looks like mark has another patch already to go.

**Comment by marktriggs** [comment]
Just noting one additional issue we found so we don't forget: unboundid treats hitting maxResultSize as an error condition and throws an exception.  Patched to return the partial result set in this case: https://github.com/NYUeServ/sakai11/commit/de48af719e04a5fd70aa70a0db27c32360aa1928

**Comment by jonespm** [comment]
@ottenhoff Do we still want to get this in? Has one conflict.

**Comment by ncaidin** [comment]
@ottenhoff do we still want to get this in?

**Comment by ottenhoff** [comment]
I'm fine closing this if there is little community interest.

**Pull request closed without merging**